### PR TITLE
Spiders are known to store prey for when they are hungry.

### PR DESCRIPTION
**Pull request by keepermanj**

### CODE REVIEW

**Comment by miki151** [comment]
Have you tested this? I believe the deathReason change won't do anything, as you only get these messages for your own minions.

Also, I think the spider will usually kill the children before you get there, so all you find are corpses. I might be wrong though.

**Comment by keepermanj** [comment]
Yeah, I did test it. The deathDescription was added to the minion messages for supporting future use. The place that actually is used right now is the change to the message generator.

The spider doesn't kill the children because they have become civilians in the spider tribe.

Have a go! :)

**Comment by keepermanj** [comment]
Would you like me to add a few baby spiders, while we are here?

File: collective.cpp [hunk 1]
```diff
@@ -787,16 +787,17 @@ void Collective::onPositionDiscovered(Position pos) {
...
 void Collective::onMinionKilled(WCreature victim, WCreature killer) {
+  string deathDescription=victim->getAttributes().getDeathDescription();
   control->onMemberKilled(victim, killer);
   if (hasTrait(victim, MinionTrait::PRISONER) && killer && getCreatures().contains(killer))
     returnResource({ResourceId::PRISONER_HEAD, 1});
   if (!hasTrait(victim, MinionTrait::FARM_ANIMAL) && !hasTrait(victim, MinionTrait::SUMMONED)) {
     decreaseMoraleForKill(killer, victim);
     if (killer)
-      control->addMessage(PlayerMessage(victim->getName().a() + " is killed by " + killer->getName().a(),
+      control->addMessage(PlayerMessage(victim->getName().a() + " is "+deathDescription+" by " + killer->getName().a(),
```
Comment (miki151) [inline collective.cpp]: Spaces around + here and elsewhere.

File: creature_factory.cpp [hunk 2]
```diff
@@ -1305,6 +1305,18 @@ CreatureAttributes CreatureFactory::getAttributesFromId(CreatureId id) {
...
+      return CATTR(
+          c.viewId = ViewId::CHILD;
+          c.attr = LIST(2_dam, 2_def );
+          c.body = Body::humanoid(Body::Size::MEDIUM);
+          c.innocent = true;
+          c.permanentEffects[LastingEffect::ENTANGLED] = 1;
+          c.permanentEffects[LastingEffect::BLIND] = 1;
+          c.chatReactionFriendly = "\"Put me out of my misery PLEASE!\""_s;
+          c.chatReactionHostile = "\"End my torture!\""_s;
+		  c.deathDescription = "dead, released from unthinkable agony"_s;
```
Comment (miki151) [inline creature_factory.cpp]: Remove tabs.

File: creature_factory.h [hunk 3]
```diff
@@ -70,6 +70,7 @@ RICH_ENUM(CreatureId,
     MINOTAUR,
     HYDRA,
     SHELOB,
+	SPIDER_FOOD,
```
Comment (miki151) [inline creature_factory.h]: Tab

File: message_generator.cpp [hunk 4]
```diff
@@ -27,7 +27,8 @@ static void addThird(WConstCreature c, MsgType type, const string& param) {
                                (param.size() > 0 ? " on the " + param : ".");
                                unseenMsg = "You hear snoring."; break;
     case MsgType::WAKE_UP: msg = c->getName().the() + " wakes up."; break;
-    case MsgType::DIE: msg = c->getName().the() + " is killed!"; break;
+    case MsgType::DIE: msg = c->getName().the() + " is "+c->getAttributes().
+	    getDeathDescription()+"!"; break;
```
Comment (miki151) [inline message_generator.cpp]: Tab

File: creature_attributes.h [hunk 5]
```diff
@@ -131,4 +133,5 @@ class CreatureAttributes {
   bool SERIAL(noAttackSound) = false;
   optional<CreatureId> SERIAL(creatureId);
   optional<double> SERIAL(moraleSpeedIncrease);
+  string SERIAL(deathDescription)="killed"_s;
```
Comment (miki151) [inline creature_attributes.h]: Make sure you add deathDescription to the serialization function.

Comment (SoftMonster) [reply creature_attributes.h]: I will have to work out how that works.

**Comment by miki151** [comment]
Check out the serialize function in creature_attributes.cpp. Every
attribute that you sorround with SERIAL needs to be listed there as well.

**Comment by SoftMonster** [comment]
Ok

**Comment by keepermanj** [comment]
Wasn't 100% sure how to test the serialization of deathMessage, so I saved the game and reloaded it. Then I killed the "food" and checked that the death message was still correct.

File: message_generator.cpp [hunk 6]
```diff
@@ -27,7 +27,8 @@ static void addThird(WConstCreature c, MsgType type, const string& param) {
                                (param.size() > 0 ? " on the " + param : ".");
                                unseenMsg = "You hear snoring."; break;
     case MsgType::WAKE_UP: msg = c->getName().the() + " wakes up."; break;
-    case MsgType::DIE: msg = c->getName().the() + " is killed!"; break;
+    case MsgType::DIE: msg = c->getName().the() + " is " + 
+                          c->getAttributes().getDeathDescription() + "!"; break;
```
Comment (miki151) [inline message_generator.cpp]: Please indent with 4 spaces as below.

File: collective.cpp [hunk 7]
```diff
@@ -787,16 +787,17 @@ void Collective::onPositionDiscovered(Position pos) {
 }
 
 void Collective::onMinionKilled(WCreature victim, WCreature killer) {
+  string deathDescription=victim->getAttributes().getDeathDescription();
```
Comment (miki151) [inline collective.cpp]: Spaces around =

**Comment by keepermanj** [comment]
Good?

**Pull request merged**